### PR TITLE
deconv cudnn forward passed

### PR DESCRIPTION
**Pull request by zchen0211**
working on backward now...

### CODE REVIEW

**Comment by chengduoZH** [comment]
You should remove  uncorrelated commits. Maybe `git cherry-pick` is helpful to you.

File: paddle/operators/conv2dtranspose_cudnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                 "better hardward. This size should be carefully setted.")
+        .SetDefault(4096);
+  }
+};
+
+}  // namespace operators
+}  // namespace paddle
+
+namespace ops = paddle::operators;
+REGISTER_OP(conv2dtranspose_cudnn, ops::Conv2DTransposeOp,
```
Comment (chengduoZH) [inline paddle/operators/conv2dtranspose_cudnn_op.cc]: Is it appropriate that replacing `conv2dtranspose_cudnn` with `conv2d_transpose_cudnn `

**Comment by chengduoZH** [comment]
`deconv cudnn` and `conv cudnn` have a lot of common parts. I think we can write these together.

File: paddle/operators/conv2dtranspose_cudnn_op.cc (hunk 1 above)
Comment (zchen0211) [reply paddle/operators/conv2dtranspose_cudnn_op.cc]: Done.

Comment (zchen0211) [reply paddle/operators/conv2dtranspose_cudnn_op.cc]: Done.

File: paddle/operators/conv2d_transpose_cudnn_op.cu [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+namespace operators {
+
+using Tensor = framework::Tensor;
+using ScopedTensorDescriptor = platform::ScopedTensorDescriptor;
+using ScopedFilterDescriptor = platform::ScopedFilterDescriptor;
+using ScopedConvolutionDescriptor = platform::ScopedConvolutionDescriptor;
+using DataLayout = platform::DataLayout;
+using CUDADeviceContext = platform::CUDADeviceContext;
+
+static constexpr size_t kCONV_CUDNN_WORKSPACE_LIMIT_BYTES = 1024 * 1024 * 1024;
```
Comment (jacquesqiao) [inline paddle/operators/conv2d_transpose_cudnn_op.cu]: this not meed the google code style.
http://google.github.io/styleguide/cppguide.html#Constant_Names

File: paddle/operators/conv2d_transpose_cudnn_op.cu [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    PADDLE_ENFORCE(platform::is_gpu_place(ctx.GetPlace()),
+                   "It must use GPUPlace.");
+    auto* input = ctx.Input<Tensor>("Input");
+    auto* filter = ctx.Input<Tensor>("Filter");
+    auto* output = ctx.Output<Tensor>("Output");
+
+    std::vector<int> strides = ctx.Attr<std::vector<int>>("strides");
+    std::vector<int> paddings = ctx.Attr<std::vector<int>>("paddings");
+    std::vector<int> dilations = ctx.Attr<std::vector<int>>("dilations");
```
Comment (jacquesqiao) [inline paddle/operators/conv2d_transpose_cudnn_op.cu]: add a comment that cuDNN does not support dilations.

File: paddle/operators/conv2d_transpose_cudnn_op.cu [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    paddle::memory::Free(gpu, cudnn_workspace);
+  }
+};
+
+}  // namespace operators
+}  // namespace paddle
+
+namespace ops = paddle::operators;
+
+REGISTER_OP_GPU_KERNEL(conv2d_transpose_cudnn,
```
Comment (jacquesqiao) [inline paddle/operators/conv2d_transpose_cudnn_op.cu]: why name it `conv2d_transpose`?

**Pull request merged**